### PR TITLE
Nonlinearly_update_residual: Ensure likelihood handling under jit

### PR DESCRIPTION
**Pull request by ph-frank**
Swap jit and likelihood insertion to enable non-static argument handling

### CODE REVIEW

File: src/re/evi.py [hunk 1]
```diff
@@ -182,20 +182,24 @@ def _metric(e, primals, tangents, *, point_estimates):
...
-    def _sampnorm(e, natgrad, *, point_estimates):
+    def _sampnorm(likelihood, e, natgrad, *, point_estimates):
         lh, e_liquid = likelihood.freeze(
             point_estimates=point_estimates, primals=e
         )
         fpp = lh.right_sqrt_metric(e_liquid, natgrad)
         return jnp.sqrt(vdot(natgrad, natgrad) + vdot(fpp, fpp))
 
+    def _trafo(likelihood, e):
+        return likelihood.transformation(e)
```
Comment (Edenhofer) [inline src/re/evi.py:193]: The usual `likelihood.freeze` shenanigans isn't necessary here as the trafo is only called once right? I think a comment would be great and maybe even a slightly different name (_trafo_wo_freeze?).

**Review by Edenhofer — COMMENTED** [review]
LGTM but I think a comment would be great (and maybe even renaming the function) on why the usual `likelihood.freeze` is not needed for the trafo here.

File: src/re/evi.py (hunk 1 above)
Comment (ph-frank) [reply src/re/evi.py:193]: Exactly. On second thought, we might also leave the `likelihood.freeze` magic in there for consistency. As you said, it is only called once per sampling run anyway.

Comment (ph-frank) [reply src/re/evi.py]: I went ahead and added the freeze also to `_trafo`. This makes the design of `_nonlinearly_update_residual_functions` self consistent and IMHO better. If you are happy with the changes this should be good to go!

**Comment by Edenhofer** [comment]
Merged! Thanks for fixing the JIT in the nonlinear sampling :)

**Pull request merged**